set event result, and add warnings to velocity plugin messaging docs (#479)
PaperMC/docs: docs/velocity/dev/api/plugin-messaging.mdx

    backend -->|"3 (Incoming)"| Velocity -->|"4 (Outgoing)"| player
```

> :::warning

Reviewer: this admonition seems out of place

Author: would it be better to be below the mention of the bungeecord plugin channel compat?

as this is probably the most important information on the page, security wise, i would prefer it to remain above the fold, as well as remain emphasized.

Reviewer: after some deliberation, I initially thought it'd be better in Case 1, as the first section is more abstract and not really about handling the messages, but then it'd need to be reiterated in Case 3, so it's probably fine as it is

you can make the admonition `danger` for additional emphasis

> When listening to PluginMessageEvent, ensure sure the result is

Reviewer: ```suggestion
When listening to `PluginMessageEvent`, ensure the result is
```

<Javadoc name={"com/velocitypowered/api/event/connection/PluginMessageEvent$ForwardResult#handled()"} project={"velocity"}>`ForwardResult.handled()`</Javadoc>
if you do not intend the client to participate.

> If the result is forwarded, Players can spoof the proxy to your backends.

Reviewer: ```suggestion
If the result is forwarded, players can impersonate the proxy to your backend servers.
```

> Additionally, ensure the result is set correct after actually handling correct messages, to prevent them from being leaked to the other party.

Reviewer: ```suggestion
Additionally, ensure the result is set correctly after actually handling correct messages, to prevent them from being leaked to the other party.
```

@Subscribe
public void onPluginMessageFromPlayer(PluginMessageEvent event) {

>     // Check if Identifier matches first, no matter the source.

Reviewer: ```suggestion
    // Check if the identifier matches first, no matter the source.
```

    if (!IDENTIFIER.equals(event.getIdentifier())) {
        return;
    }

    // mark PluginMessage as handled, indicating that the contents
    // should not be forwarding to their original destination.
    event.setResult(PluginMessageEvent.ForwardResult.handled());

    // Alternatively:

    // mark PluginMessage as forwarded, indicating that the contents

>     // should be passed through, as if velocity is not present.

Reviewer: ```suggestion
    // should be passed through, as if Velocity is not present.
```

@Subscribe
public void onPluginMessageFromBackend(PluginMessageEvent event) {

>     // Check if Identifier matches first, no matter the source.

Reviewer: ```suggestion
    // Check if the identifier matches first, no matter the source.
```
